### PR TITLE
Best buy/sdkversion 24

### PR DESCRIPTION
**Pull request by PetrosZip**
Update SDK to min version 24, but had to do updates on gradle and other configuration updates.

### CODE REVIEW

**Review by rhyswilliamszip — DISMISSED** [review]

File: quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java [hunk 1]
```diff
@@ -287,27 +288,17 @@ public void WidgetDefault(SpannableStringBuilder sb, VerticalImageSpan imageSpan
...
-        }
 
         if (amount == null || amount.equals("")) {
 
-            amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(minOrder)));
+            amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(min.toString())));
 
         } else {
-            if (Float.parseFloat(amount) < Float.parseFloat(minOrder))
+            if (Float.parseFloat(amount) < Float.parseFloat(min.toString()))
```
Comment (rhyswilliamszip) [inline quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java:297]: We're converting an integer to a string to then parse it to float.  I'm pretty sure Java will do the automatic conversion to compare the float to an integer.  Can we change all locations where this is happening.

File: quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java [hunk 2]
```diff
@@ -287,27 +288,17 @@ public void WidgetDefault(SpannableStringBuilder sb, VerticalImageSpan imageSpan
...
             {
-                amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(minOrder)));
-            } else if (Float.parseFloat(amount) > Float.parseFloat(maxOrder)) {
-                amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(maxOrder)));
+                amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(min.toString())));
+            } else if (Float.parseFloat(amount) > Float.parseFloat(max.toString())) {
+                amountString = new SpannableString(" $" + decimalFormat.format(Float.parseFloat(max.toString())));
```
Comment (rhyswilliamszip) [inline quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java:301]: Again no need to `parseFloat` and `toString` when the value is already an integer

Comment (PetrosZip) [reply quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java]: Done

File: quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java (hunk 1 above)
Comment (PetrosZip) [reply quadpay/src/main/java/com/quadpay/quadpay/Widget/QuadPayWidgetTextView.java]: Done

**Review by rhyswilliamszip — APPROVED** [review]

**Pull request merged**